crates/bevy_asset/src/processor/mod.rs

```diff
@@ -1,3 +1,25 @@
+//! Asset processing in Bevy is a framework by which artist-authored assets can be automatically prepared for use in game.
+//!
+//! You can think of the asset processing system as a "build system" for assets.
+//! When an artist adds a new asset to the project, the asset processing system will automatically perform the specified processing steps on the asset.
+//! This can include things like converting a `.psd` file to a `.png`, compressing a `.wav` file to an `.ogg`, or generating mipmaps for a texture.
+//!
+//! Its core values are:
+//!
+//! 1. Automatic: changing the artist-format asset should not require any manual steps to get it back in the game.
+//! 2. Configurable: every game has its own needs, and a high level of transparency and control is required.
+//! 3. Lossless: the original asset should always be preserved, avoiding frustration and making experimentation safe.
+//! 4. Deterministic: performing the same processing steps on the same asset should always produce the same result.
```

Comment:
Ehh idk if we want to guarantee determinism. For meshlets it's not true on windows due to a global PRNG we don't control in metis 😓

Reply:
Hmm. I think it's a desirable property in general, even if it can't rise to the level of a guarantee. Clearly calling out that meshlet preprocessing is not deterministic in its docs feels correct to me, which suggests that this is an important assumption that users are going to make / want.

Reply:
I don't disagree, but it might be hard for rendering. Stuff like lightmap baking is never going to be derterministic idt, as it's inherently an estimate + denoiser of lighting.

Reply:
"should" is already sufficiently weak wording I think, but you could work around it with an "attempt" or "interchangeable" weasel word. I think noting that an asset processor that, say, generates random data or uses a web api or similar is a bad idea is important though, to steer people away from using it as a generic caching system for computations. Even if some assets may not be entirely deterministic in practice. However the next sentence clarifies why that is anyway.

Reply:
"should" is already sufficiently weak wording I think, but you could work around it with an "attempt" or "interchangeable" weasel word. I think noting that an asset processor that, say, generates random data or uses a web api or similar is a bad idea is important though, to steer people away from using it as a generic caching system for computations. Even if some assets may not be entirely deterministic in practice. However the next sentence clarifies why that is anyway.

Reply:
Yeah I think its worth making this something like:

```suggestion
//! 4. Deterministic: performing the same processing steps on the same asset should (generally) produce the exact same result. In cases where this doesn't make sense (steps that involve a degree of randomness or uncertainty), the results across runs should be "acceptably similar", as they will be generated once for a given set of inputs and cached.
```